jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/S3ViewerProvider.kt

```diff
@@ -74,12 +75,20 @@ class S3ViewerEditor(project: Project, bucket: S3VirtualBucket) : UserDataHolder
     override fun setState(state: FileEditorState) {}
 }
 
-fun openEditor(project: Project, bucket: Bucket) {
-    try {
-        FileEditorManager.getInstance(project).openTextEditor(OpenFileDescriptor(project, S3VirtualBucket(bucket, project.awsClient())), true)
-        S3Telemetry.openEditor(project, Result.SUCCEEDED)
-    } catch (e: Exception) {
-        e.notifyError(message("s3.open.viewer.bucket.failed"))
-        S3Telemetry.openEditor(project, Result.FAILED)
-    }
+fun openEditor(project: Project, bucket: Bucket): Editor? = try {
+    val editor = FileEditorManager.getInstance(project).openTextEditor(
```

Comment:
I've found myself in the habit of using `also` instead of assigning to a variable - so something like `FileEditorManager.getInstance(..).openTextEditor(...).also { S3Telemetry.openEditor(SUCCESS) }`

Reply:
ya i went back and forth. I'll go with also since its 1 ine